types/webpack/index.d.ts

```diff
@@ -23,2502 +23,8 @@
 //                 Piotr Błażejewicz <https://github.com/peterblazejewicz>
 //                 Michał Grzegorzewski <https://github.com/spamshaker>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.3
+// TypeScript Version: 3.7
 
 /// <reference types="node" />
-
-import { Hash as CryptoHash } from 'crypto';
-import {
-  Tapable,
-  HookMap,
-  SyncBailHook,
-  SyncHook,
-  SyncLoopHook,
-  SyncWaterfallHook,
-  AsyncParallelBailHook,
-  AsyncParallelHook,
-  AsyncSeriesBailHook,
-  AsyncSeriesHook,
-  AsyncSeriesWaterfallHook,
-} from 'tapable';
-import * as UglifyJS from 'uglify-js';
-import * as anymatch from 'anymatch';
-import { RawSourceMap } from 'source-map';
-import { Source, ConcatSource } from 'webpack-sources';
-
+import webpack = require('./node_modules/webpack');
```

Comment:
🥇

Reply:
@sandersn Hm, am I missing something, or will this just not work in a monorepo that uses yarn workspaces, and hoists its dependencies to the root `node_modules`? Webpack can not be found at this path in this scenario. Instead it's located at `'../../webpack'` relative to the location of `node_modules/@types/webpack/index.d.ts`. So the following error will occur:

```
error TS2307: Cannot find module './node_modules/webpack' or its corresponding type declarations.
```

Reply:
If I change this line to `import webpack = require('webpack');`, it does work. Is this not an option because it would require the tsconfig option `"moduleResolution": "node"`?

Reply:
It doesn't work in the DT repo because it's a circular reference...maybe if I add a path mapping in the tsconfig...